### PR TITLE
补全PWM设备，并为每个PWM设备添加条件编译，减少代码量

### PR DESCRIPTION
**Pull request by muaxiaohei**
## 拉取/合并请求描述：(PR description)

[
补全PWM设备，并为每个PWM设备添加条件编译，减少代码量
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    rt_uint8_t channel[4];
+    char* name;
+};
+
+
+
+
+
+
+
```
Comment (mysterywolf) [inline bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.h:44]: 请注意这些多余的空格 只留一个空行即可

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.c [hunk 2]
```diff
@@ -1,628 +1,963 @@
...
+    case PWM_CMD_ENABLE:
+        return ch32_pwm_device_enable(device, configuration, RT_TRUE);
+    case PWM_CMD_DISABLE:
+        return ch32_pwm_device_enable(device, configuration, RT_FALSE);
+    case PWM_CMD_SET:
+        return ch32_pwm_device_set(device, configuration);
+    case PWM_CMD_GET:
+        return ch32_pwm_device_get(device, configuration);
+    default:
+        return RT_EINVAL;
```
Comment (mysterywolf) [inline bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.c:921]: 请注意 RT-Thread的错误码除了RT_EOK其他都是需要加负号的

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.h (hunk 1 above)
Comment (muaxiaohei) [reply bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.h]: 已根据建议进行修改

**Review by muaxiaohei — COMMENTED** [review]
已根据建议修改

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#ifndef __DRV_PWM_H__
+#define __DRV_PWM_H__
+
+#include "ch32v30x_tim.h"
+#include <rtthread.h>
+#include <drivers/rt_drv_pwm.h>
+#include <drivers/hwtimer.h>
+#include <board.h>
+
+#ifdef BSP_USING_PWM
```
Comment (mysterywolf) [inline bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.h:20]:
```suggestion
#include <rtthread.h>
#ifdef BSP_USING_PWM
#include "ch32v30x_tim.h"
#include <drivers/rt_drv_pwm.h>
#include <drivers/hwtimer.h>
#include <board.h>
```

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.c [hunk 4]
```diff
@@ -1,628 +1,963 @@
...
+ */
+
+#include "drv_pwm.h"
+
+#ifdef BSP_USING_PWM
+
+#define LOG_TAG "drv.pwm"
+#include <drv_log.h>
+
+
```
Comment (mysterywolf) [inline bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.c:20]:
```suggestion

#ifdef BSP_USING_PWM

#define LOG_TAG "drv.pwm"
#include <drv_log.h>

```

File: bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.c [hunk 5]
```diff
@@ -1,628 +1,963 @@
...
+    RCC_GetClocksFreq(&RCC_Clocks);
+
+    /*tim1~10 all in HCLK*/
+    return RCC_Clocks.HCLK_Frequency;
+}
+
+/*
+ * NOTE:    some pwm pins of some timers are reused,
+ *          please keep caution when using pwm
+ *  */
```
Comment (mysterywolf) [inline bsp/wch/risc-v/Libraries/ch32_drivers/drv_pwm.c:94]:
```suggestion
/*
 * NOTE:  some pwm pins of some timers are reused,
 *          please keep caution when using pwm
 */

```

**Review by mysterywolf — CHANGES_REQUESTED** [review]

**Review by mysterywolf — APPROVED** [review]

**Pull request merged**